src/Microsoft.AspNetCore.SystemWebAdapters/Hosting/HostingEnvironmentAccessor.cs

```diff
@@ -2,20 +2,25 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.SystemWebAdapters;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Options;
 
 namespace System.Web.Hosting;
 
 internal sealed class HostingEnvironmentAccessor
 {
+    private static readonly HttpContextAccessor _defaultHttpContextAccessor = new();
```

Comment:
> how often is _accessor or _current null

good point - most of the time they are not null.  It will only be if someone is running outside of a asp.net core host. I've changed it to just be a back up.